dashi/bcl2fastq/index_summary.py

```diff
@@ -27,75 +30,83 @@
  "Count","LaneNumber","Index1","Index2","Run","LIMS IUS SWID
  """
 
-"""Within the main layout division there are 4 groups: 1. the section with the
-100% bar graph 2.  Nested subdivision of: a) pie graph at 24% b) bar graph at
-76% 3. Sample Run hidden 4. Pruned  Unknown Hidden Note: span can be used for
-inline setting
-
-
-Dropdown option's label set to all_runs(combination of unknown and
-known) Dropdown multi set to false (default) - only single select
-possible -  multiple select gives error due to inconsistent values """
-layout = html.Div(
-    children=[
-        dcc.ConfirmDialog(
-            id="error",
-            message=(
-                'You have input an incorrect run. Click either "Ok" or '
-                '"Cancel" to return to the most recent run.'
-            ),
-        ),
-        dcc.Dropdown(
-            id="run_select",
-            #   Options is concantenated string versions of all_runs.
-            options=[{"label": r, "value": r} for r in all_runs],
-            value=all_runs[0],
-            clearable=False,
-        ),
-        # This element doesn't work correctly in a multi-app context. Left in code for further work
-        # ToDO
-        # https://jira.oicr.on.ca/browse/GR-776 and https://jira.oicr.on.ca/browse/GR-777
-        dcc.Location(id="bcl2fastq_url", refresh=False),
-        dcc.Graph(id="known_index_bar"),
-        html.Div(
-            [
-                html.Div(
-                    [
-                        dcc.Graph(id="known_unknown_pie"),
-                        dcc.Textarea(
-                            id="known_fraction",
-                            style={"width": "100%"},
-                            readOnly=True,
-                            # This is the textbox at the bottom, hover over to see title
-                            title=(
-                                "Assumptions are made about which indexes are known "
-                                "or unknown. This is due to multiple bcl2fastq analyses "
-                                "being used on one run. This number should be 100%."
-                            ),
-                        ),
-                    ],
-                    style={"width": "25%", "display": "inline-block"},
-                ),
-                html.Div(
-                    [dcc.Graph(id="unknown_index_bar")],
-                    style={
-                        "width": "75%",
-                        "display": "inline-block",
-                        "float": "right",
-                    },
-                ),
-            ]
-        ),
-    ]
-)
 
-try:
-    from app import app
-except ModuleNotFoundError:
-    import dash
+def generate_layout(qs) -> html.Div:
```

Comment:
Contents of 'qs' variable was unclear to me. Had to dig into the code to figure out it's the search term grabbed from the URL. Documentation of source of 'qs' (and why the first character must be removed?) would be appreciated